app/controllers/lists_controller.rb

```diff
@@ -0,0 +1,68 @@
+class ListsController < ApplicationController
+  before_action :authenticate_user!
+
+  def show
+    @list = List.find(params[:id])
+    authorize @list
+  end
+
+  def index
+    @lists = current_user.lists
+    authorize @lists
+  end
+
+  def new
+    @list = List.new
+    authorize @list
+  end
+
+  def edit
+    @list = List.find(params[:id])
+    authorize @list
+  end
+
+  def create
+    @list = current_user.lists.build(list_params)
+    authorize @list
+
+    if @list.save
+      flash[:notice] = "List was saved."
+      redirect_to @list
+    else
+      flash[:error] = "There was an error saving the list. Please try again."
+      render :new
+    end
+  end
+
+  def update
```

Comment:
Write a controller spec that has 2 users: `attacker` and `victim`. Log the `attacker` in and send an update request to update the `victim`'s list title.

Also write a spec where the logged-in `attacker` deletes the `victim`'s list.

Once these are failing as expected, then fix the controller by using scoped queries like:

```ruby
@list = current_user.lists.find(params[:id])
```